src/libraries/System.Private.Xml/src/System/Xml/Serialization/Types.cs

```diff
@@ -1239,6 +1281,7 @@ private static TypeFlags GetConstructorFlags(Type type, ref Exception? exception
             return 0;
         }
 
+        [RequiresUnreferencedCode("creates XmlAttributes")]
```

Comment:
Does this still need to be `RequiresUnreferencedCode` now that `XmlAttributes` is no longer marked as unsafe?

If it doesn't, i think a bunch of other stuff could be unmarked as well, like `ImportTypeDesc` and `GetTypeDesc`, which means other things, maybe even that static ctor issue could just go away.

Reply:
Unfortunately not that much benefit. Only GetAttributes seemed clean, and all the uses of that ended up needing RUC for some other reason.

Reply:
Oh well... it was a good try though.